cmake/macos/compilerconfig.cmake

```diff
@@ -30,40 +30,98 @@ set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 
 # Add default C and C++ compiler options if Xcode generator is not used
 if(NOT XCODE)
-  list(
-    APPEND
-    _obs_c_options
-    -Werror
-    -Wextra
-    -Wvla
-    -Wswitch
-    -Wformat-security
-    -Wunused-parameter
-    -Wno-unused-function
-    -Wno-missing-field-initializers
-    -Wformat
-    -fno-strict-aliasing
-    -Wno-error=shorten-64-to-32)
+  set(_obs_c_options
+      # cmake-format: sortable
+      -fno-strict-aliasing
+      -Wbool-conversion
+      -Wcomma
+      -Wconstant-conversion
+      -Wdeprecated-declarations
+      -Wempty-body
+      -Wenum-conversion
+      -Werror
+      -Werror=block-capture-autoreleasing
+      -Werror=return-type
+      -Wextra
+      -Wformat
+      -Wformat-security
+      -Wfour-char-constants
+      -Winfinite-recursion
+      -Wint-conversion
+      -Wnewline-eof
+      -Wno-conversion
+      -Wno-float-conversion
+      -Wno-implicit-fallthrough
+      -Wno-missing-braces
+      -Wno-missing-field-initializers
+      -Wno-missing-prototypes
+      -Wno-semicolon-before-method-body
+      -Wno-shadow
+      -Wno-sign-conversion
+      -Wno-trigraphs
+      -Wno-unknown-pragmas
+      -Wno-unused-function
+      -Wno-unused-label
+      -Wnon-literal-null-conversion
+      -Wobjc-literal-conversion
+      -Wparentheses
+      -Wpointer-sign
+      -Wquoted-include-in-framework-header
+      -Wshorten-64-to-32
+      -Wsign-compare
+      -Wstrict-prototypes
+      -Wswitch
+      -Wuninitialized
+      -Wunreachable-code
+      -Wunused-parameter
+      -Wunused-value
+      -Wunused-variable
+      -Wvla)
+
+  set(_obs_cxx_options
+      # cmake-format: sortable
+      -Warc-repeated-use-of-weak
+      -Wconversion
+      -Wdeprecated-implementations
+      -Wduplicate-method-match
+      -Wfloat-conversion
+      -Wfour-char-constants
+      -Wimplicit-retain-self
+      -Winvalid-offsetof
+      -Wmove
+      -Wno-arc-maybe-repeated-use-of-weak
+      -Wno-c++11-extensions
```

Comment:
Out of curiosity, why do we disable this warning? Shouldn't setting `CMAKE_CXX_EXTENSIONS OFF` (like we currently do) prevent us from being able to disable warnings about using C++11 extensions? Also, don't we *want* to avoid using extensions, so shouldn't we want the warnings? I'd actually almost rather we use `-Wpedantic`, since my understanding was that we want to avoid using extensions at all.

Reply:
I just explicitly added the warnings that Xcode enables by default so that compilers complain about the same things.

Reply:
This is disabling a warning though, unless I'm mistaken?

Reply:
That is correct - I just enabled the preferred project settings and copied the compiler flags. I wonder if this one has any effect given that we enforce C++17 anyway, so I wouldn't be too fussed about removing it.